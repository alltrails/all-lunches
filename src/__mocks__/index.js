export { default as restaurantMock } from './restaurant.mock';
export { default as restaurantsMock } from './restaurants.mock';
export { default as reactRefMock } from './reactRef.mock';
