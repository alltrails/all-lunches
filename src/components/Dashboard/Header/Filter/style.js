import styled from 'styled-components';
import TransparentButton from 'styles/components/TransparentButton';
import { colors } from 'styles';

export const Wrapper = styled.div`
  position: relative;
`;

export const FilterMenuButton = styled(TransparentButton)`
  background: ${(p) => (p.isMenuOpen ? colors.green : colors.white)};
  font-size: 1.2rem;
  padding: 0 14px;
  color: ${(p) => (p.isMenuOpen ? colors.white : colors.grayDark)};
  border: 1px solid ${colors.borderGray};
  border-radius: 6px;
  margin-right: 16px;
  height: 100%;
  min-width: 58px;

  &.focus-visible,
  &:hover {
    border: 1px solid ${colors.green};
    border-radius: 6px;
  }
`;
