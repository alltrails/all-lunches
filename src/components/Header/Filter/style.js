import styled from 'styled-components';
import TransparentButton from 'styles/components/TransparentButton';
import { colors } from 'styles';

export const Button = styled(TransparentButton)`
  height: 100%;
  width: 52px;
  margin-right: 16px;
  border: 1px solid ${colors.borderGray};
  border-radius: 6px;
  color: ${colors.baseGray};
  background: white;
  font-size: 1.2rem;
`;
