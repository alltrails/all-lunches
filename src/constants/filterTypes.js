export const HIGH_TO_LOW_FILTER_OPTION = 'high-to-low';
export const LOW_TO_HIGH_FILTER_OPTION = 'low-to-high';
