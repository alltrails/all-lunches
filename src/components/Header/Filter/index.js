import React from 'react';

import { Button } from './style';

const Filter = () => {
  console.log('test');

  return <Button>Filter</Button>;
};

export default Filter;
