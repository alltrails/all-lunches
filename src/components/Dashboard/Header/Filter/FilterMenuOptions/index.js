import React, { useRef } from 'react';
import PropTypes from 'prop-types';

import { refType } from 'constants/propTypes';
import * as Types from 'constants/filterTypes';

import { useEventListener } from 'lib/customHooks';

import CheckmarkIcon from 'components/svg/CheckmarkIcon';
import FocusTrap from 'components/shared/FocusTrap';
import {
  ApplyButton,
  CheckmarkIconWrapper,
  Form,
  ListWrapper,
  MenuOptions,
  OptionItem,
  OptionItemButton,
} from './style';

const menuOptions = [
  { id: Types.HIGH_TO_LOW_FILTER_OPTION, title: 'Ratings High to Low' },
  { id: Types.LOW_TO_HIGH_FILTER_OPTION, title: 'Ratings Low to High' },
];

const FilterMenuOptions = ({
  filterButtonRef,
  onClose,
  onMenuOptionClick,
  onSubmit,
  selectedMenuOption,
}) => {
  const menuRef = useRef();

  const handleMouseDown = ({ target }) => {
    const shouldCloseMenu =
      !menuRef.current?.contains(target) && !filterButtonRef.current?.contains(target);

    if (shouldCloseMenu) onClose();
  };

  useEventListener('mousedown', handleMouseDown);

  return (
    <FocusTrap fallbackFocus="#pop-out-menu">
      <MenuOptions ref={menuRef} tabIndex="-1" id="pop-out-menu">
        <Form>
          <ListWrapper>
            {menuOptions.map((option) => {
              const isSelected = selectedMenuOption === option.id;

              return (
                <OptionItemButton
                  key={option.id}
                  onClick={() => onMenuOptionClick(option.id)}
                  type="button"
                >
                  <OptionItem>
                    <CheckmarkIconWrapper isSelected={isSelected}>
                      <CheckmarkIcon
                        titleText={option.title}
                        variant={isSelected ? 'green' : 'white'}
                      />
                    </CheckmarkIconWrapper>
                    {option.title}
                  </OptionItem>
                </OptionItemButton>
              );
            })}
          </ListWrapper>
          <ApplyButton disabled={!selectedMenuOption} onClick={onSubmit}>
            Apply
          </ApplyButton>
        </Form>
      </MenuOptions>
    </FocusTrap>
  );
};

FilterMenuOptions.propTypes = {
  filterButtonRef: refType,
  onMenuOptionClick: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  selectedMenuOption: PropTypes.string,
  onClose: PropTypes.func.isRequired,
};

FilterMenuOptions.defaultProps = {
  filterButtonRef: null,
  selectedMenuOption: null,
};

export default FilterMenuOptions;
