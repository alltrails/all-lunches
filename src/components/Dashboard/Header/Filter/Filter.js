import React from 'react';
import PropTypes from 'prop-types';

import { refType } from 'constants/propTypes';

import { FilterMenuButton, Wrapper } from './style';
import FilterMenuOptions from './FilterMenuOptions';

const Filter = ({
  isMenuOpen,
  filterButtonRef,
  onMenuOptionClick,
  onSubmit,
  selectedMenuOption,
  toggleMenuOpen,
}) => (
  <Wrapper>
    <FilterMenuButton ref={filterButtonRef} isMenuOpen={isMenuOpen} onClick={toggleMenuOpen}>
      {isMenuOpen ? 'Sort' : 'Filter'}
    </FilterMenuButton>
    {isMenuOpen && (
      <FilterMenuOptions
        filterButtonRef={filterButtonRef}
        onClose={() => toggleMenuOpen(true)}
        onMenuOptionClick={onMenuOptionClick}
        onSubmit={onSubmit}
        selectedMenuOption={selectedMenuOption}
      />
    )}
  </Wrapper>
);

Filter.propTypes = {
  filterButtonRef: refType,
  isMenuOpen: PropTypes.bool.isRequired,
  onMenuOptionClick: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  selectedMenuOption: PropTypes.string,
  toggleMenuOpen: PropTypes.func.isRequired,
};

Filter.defaultProps = {
  filterButtonRef: null,
  selectedMenuOption: null,
};

export default Filter;
