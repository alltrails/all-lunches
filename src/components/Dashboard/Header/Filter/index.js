import React, { useRef, useState } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import * as restaurantsActions from 'store/restaurants/actions';

import Filter from './Filter';

const mapDispatchToProps = {
  setFilterOption: restaurantsActions.setFilterOption,
};

const FilterContainer = ({ setFilterOption }) => {
  const filterButtonRef = useRef();

  const [isMenuOpen, setIsMenuOpen] = useState(false);
  const [selectedMenuOption, setSelectedMenuOption] = useState(null);

  const handleMenuOptionClick = (menuOptionId) => {
    setSelectedMenuOption(menuOptionId);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (selectedMenuOption) setFilterOption(selectedMenuOption);
  };

  return (
    <Filter
      isMenuOpen={isMenuOpen}
      filterButtonRef={filterButtonRef}
      onMenuOptionClick={handleMenuOptionClick}
      onSubmit={handleSubmit}
      selectedMenuOption={selectedMenuOption}
      toggleMenuOpen={() => setIsMenuOpen((isOpen) => !isOpen)}
    />
  );
};

FilterContainer.propTypes = {
  setFilterOption: PropTypes.func.isRequired,
};

export default connect(null, mapDispatchToProps)(FilterContainer);
