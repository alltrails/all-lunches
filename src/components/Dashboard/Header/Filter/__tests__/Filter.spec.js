/* eslint-env jest */

import shallowSetup from 'lib/testHelpers/shallowSetup';
import * as FilterTypes from 'constants/filterTypes';
import { reactRefMock } from '__mocks__';

import Filter from '../Filter';

describe('Filter', () => {
  const defaultProps = {
    filterButtonRef: reactRefMock,
    isMenuOpen: false,
    onMenuOptionClick: () => {},
    onSubmit: () => {},
    selectedMenuOption: null,
    toggleMenuOpen: () => {},
  };
  const setup = shallowSetup(Filter, defaultProps);

  it('renders the expected view', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders the expected view when menu is open', () => {
    const { wrapper } = setup({ isMenuOpen: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders the expected view when menu is open and option is selected', () => {
    const { wrapper } = setup({
      isMenuOpen: true,
      selectedMenuOption: FilterTypes.LOW_TO_HIGH_FILTER_OPTION,
    });
    expect(wrapper).toMatchSnapshot();
  });
});
