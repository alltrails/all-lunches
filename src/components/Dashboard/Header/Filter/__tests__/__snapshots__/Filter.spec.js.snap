// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter renders the expected view 1`] = `
<style__Wrapper>
  <style__FilterMenuButton
    isMenuOpen={false}
    onClick={[Function]}
  >
    Filter
  </style__FilterMenuButton>
</style__Wrapper>
`;

exports[`Filter renders the expected view when menu is open 1`] = `
<style__Wrapper>
  <style__FilterMenuButton
    isMenuOpen={true}
    onClick={[Function]}
  >
    Sort
  </style__FilterMenuButton>
  <FilterMenuContainer
    filterButtonRef={
      Object {
        "current": null,
      }
    }
    onClose={[Function]}
    onMenuOptionClick={[Function]}
    onSubmit={[Function]}
    selectedMenuOption={null}
  />
</style__Wrapper>
`;

exports[`Filter renders the expected view when menu is open and option is selected 1`] = `
<style__Wrapper>
  <style__FilterMenuButton
    isMenuOpen={true}
    onClick={[Function]}
  >
    Sort
  </style__FilterMenuButton>
  <FilterMenuContainer
    filterButtonRef={
      Object {
        "current": null,
      }
    }
    onClose={[Function]}
    onMenuOptionClick={[Function]}
    onSubmit={[Function]}
    selectedMenuOption="low-to-high"
  />
</style__Wrapper>
`;
